rust/noosphere-gateway/src/handlers/v0alpha2/push.rs

```diff
@@ -64,16 +60,19 @@ where
             &generate_capability(counterpart.as_str(), SphereAbility::Push),
         )
         .await?;
-
     let gateway_push_routine = GatewayPushRoutine {
         gateway_sphere,
         gateway_scope,
         syndication_tx,
         name_system_tx,
-        block_stream: Box::pin(from_car_stream(stream)),
+        // In Axum 0.7+, there are `Sync` bounds required. The incoming stream
+        // is `!Sync`, but as the only consumers of the stream,
+        // consider it `Sync` via `UnsharedStream`.
+        block_stream: Box::pin(from_car_stream(UnsharedStream::new(
```

Comment:
Is there a particular reason why `from_car_stream` is on the "outside" of the `UnsharedStream`?

Reply:
No, other than the changes moving from the now-gone `axum::extract::BodyStream` to the newer `axum::Body`